### PR TITLE
Add nucleo-g431kb board

### PR DESCRIPTION
**Pull request by mcbridejc**
I needed a board definition for the nucleo-32 g431kb development board. 

If you are interested in expanding the BSP collection, feel free to merge. If not, feel free to reject and I can just include into my project. 

There are a couple pins that differ from the usual nucleo-32 pinouts (as defined in `nucleo32_arduino.hpp`) on this board

### CODE REVIEW

File: src/modm/board/nucleo_g431kb/module.lb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+def build(env):
+    env.outbasepath = "modm/src/modm/board"
+    env.substitutions = {
+        "with_logger": True,
+        "with_assert": env.has_module(":architecture:assert")
+    }
+    env.template("../board.cpp.in", "board.cpp")
+    env.copy('.')
+    env.copy("../nucleo32_arduino.hpp", "nucleo32_arduino.hpp")
```
Comment (salkinium) [inline src/modm/board/nucleo_g431kb/module.lb]:
```suggestion
```

File: src/modm/board/nucleo_g431kb/board.hpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2019, Raphael Lehmann
```
Comment (salkinium) [inline src/modm/board/nucleo_g431kb/board.hpp]: Your name.

File: src/modm/board/nucleo_g431kb/board.hpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+			.pllR = 2,	// 336MHz / R= 2 -> 170MHz = F_cpu
+		};
+		Rcc::enablePll(Rcc::PllSource::InternalClock, pllFactors);
+		Rcc::setFlashLatency<Frequency>();
+		// switch system clock to PLL output
+		Rcc::enableSystemClock(Rcc::SystemClockSource::Pll);
+		Rcc::setAhbPrescaler(Rcc::AhbPrescaler::Div1);
+		// APB1 has max. 170MHz
+		Rcc::setApb1Prescaler(Rcc::Apb1Prescaler::Div1);
+		Rcc::setApb2Prescaler(Rcc::Apb2Prescaler::Div1);
```
Comment (salkinium) [inline src/modm/board/nucleo_g431kb/board.hpp:103]: Is this correct? One of these typically has a lower max frequency, usually ≤f_cpu/2.

**Review by salkinium — CHANGES_REQUESTED** [review]
Thanks, I'll gladly merge with these minor changes.

(Please squash into one commit)

**Comment by salkinium** [comment]
> There are a couple pins that differ from the usual nucleo-32 pinouts

Yes, I've been bitten by this too already. We really need to extend modm-devices with the nucleo board pinouts and generate this…

File: src/modm/board/nucleo_g431kb/board.hpp (hunk 3 above)
Comment (mcbridejc) [reply src/modm/board/nucleo_g431kb/board.hpp:103]: I believe it is. Datasheet refers to them as low-speed and high-speed APB buses, but says max frequency on both is 170MHz.

Comment (rleh) [reply src/modm/board/nucleo_g431kb/board.hpp:103]: CubeMX says yes:

grafik

**Review by rleh — APPROVED** [review]
Thanks! :+1:

**Review by salkinium — APPROVED** [review]
Thanks!

**Comment by salkinium** [comment]
I added a GpioUnused Button and an example, so it becomes part of the CI and doesn't get lost in future refactorings.

**Pull request merged**